src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -231,7 +231,7 @@
 
         if (file != null) {
             Path basePath = FilePathService.getCourseIconFilePath();
-            Path savePath = fileService.saveFile(file, basePath);
+            Path savePath = fileService.saveFile(file, basePath, false);
```

Comment:
1. Valid comment, but it was the same before. And changing this would require quite some changes in the tests which is out of scope for now. It's an edge case and the worst that could happen would be people getting an error and after reloading it's gone.
2. The url gets automatically generated.
3. Fair, but out of scope
4. These are changes keeping the same features and just changing their implementation. Thus the tests should work the same. The only reason I touched the tests in the first place was because I now need to save the files for the tests at the correct location for the FilePathService to work.